schema/python/manifest.py

```diff
@@ -0,0 +1,163 @@
+
+class VersionDependency(AbstractDependency):
+        """ Exact version dependency
+
+        """
+        class Options(object):
+                definition_id = 'version-dependency'
+
+        type = jsl.StringField(enum=['varsion-dependency'])
+        version = jsl.DocumentField(Version, as_ref=True, required=True, description='exact version dependency')
+
+class NotVersionDependency(AbstractDependency):
+        """ Negates the result of another `AbstractDependency`
+
+        """
+        class Options(object):
+                definition_id = 'not-version-dependency'
+
+        type = jsl.StringField(enum=['not-version-dependency'])
+        target = jsl.DocumentField(AbstractDependency, as_ref=True, required=True, description='version specificaiton to negate')
+
+class AndDependencyGroup(AbstractDependency):
+        """ Collates `items` under `AND` clause
+
+        """
+        class Options(object):
+                definition_id = 'and-version-dependency'
+
+        type = jsl.StringField(enum=['and-version-dependency'])
+        items = jsl.ArrayField(jsl.DocumentField(AbstractDependency, as_ref=True),
+                  required=True, description='items to join with AND')
+
+class OrDependencyGroup(AbstractDependency):
+        """ Collates `items` under `OR` clause
+
+        """
+        class Options(object):
+                definition_id = 'or-version-dependency'
+
+        type = jsl.StringField(enum=['or-version-dependency'])
+        items = jsl.ArrayField(jsl.DocumentField(AbstractDependency, as_ref=True),
+                  required=True, description='items to join with OR')
+
+class FileTrait(jsl.Document):
+        """ Describes basic properties of file provides
+
+        """
+        class Options(object):
+                definition_id = 'file-trait'
+
+        is_executable = jsl.BooleanField(description='true for executable binaries')
+        is_library = jsl.BooleanField(description='true for library binaries')
+        is_header = jsl.BooleanField(description='true for header files')
+        is_source = jsl.BooleanField(description='true for source files')
+        language = jsl.StringField(description='implementation language', required=True)
+
+class FileProvides(jsl.Document):
+        """ Each file should corresopond to exactly one provide
+
+        """
+        class Options(object):
+                definition_id = 'file-provide'
+
+        name = jsl.StringField(description='basename of the file')
+        uri = jsl.StringField(format='uri', description='chroot relative uri of the file')
+        hash = jsl.StringField(pattern='^[A-Fa-f0-9]{32}$', description='sha256 digest of the file')
+        architecture = jsl.StringField(pattern=NAME_PATTERN, description='architecture description string')
+        traits = jsl.ArrayField(
+                   jsl.DocumentField(FileTrait, as_ref=True),
+                   description='traits of the file')
+
+class Manifest(jsl.Document):
```

Comment:
Missing items:
* License array
* License object
* Archive(s) name(s)
  * if some ecosystems have multi-file packages
* Digest of archive(s)